### ISSUE TITLE
__wasm_call_ctors regression in STANDALONE mode

### ISSUE THREAD

**Issue by zeux** [emscripten-core/emscripten#9767]
#9672 makes the following change:
> Don't export __wasm_call_ctors as we call it from _start on the inside.

This breaks STANDALONE mode. Specifically, in STANDALONE mode since _start is not present, it is now impossible to call global constructors by invoking __wasm_call_ctors. I'd like to request that some symbol is exported for this, either __wasm_call_ctors or something else with emscripten prefix.

Ideally the symbol would only get exported if any global ctors exist, but using the previous behavior where it's always exported is better than not exporting anything.

**Comment by sbc100** [comment]
IIRC in STANDALONE_WASM mode _start is *always* present.  It is defined in `system/lib/libc/crt1.c` which always get linked it.   Even if your program doesn't have a main you still get `_start` exported and called that will in turn call `__wasm_call_ctors`.

See: https://github.com/emscripten-core/emscripten/blob/c579fcd5ef78cabd19f6cee9a513894ce4e555d9/system/lib/libc/crt1.c#L34

**Comment by zeux** [comment]
This definitely doesn't happen for me.

Command line:
> emcc src/vertexcodec.cpp src/indexcodec.cpp -O3 -DNDEBUG -s EXPORTED_FUNCTIONS='["_meshopt_decodeVertexBuffer", "_meshopt_decodeIndexBuffer", "_sbrk"]' -s ALLOW_MEMORY_GROWTH=1 -s TOTAL_STACK=24576 -s TOTAL_MEMORY=65536 -o build/meshopt_decoder.wasm -munimplemented-simd128 -mbulk-memory

Exports:
```
Export[4]:
 - memory[0] -> "memory"
 - func[6] <meshopt_decodeVertexBuffer> -> "meshopt_decodeVertexBuffer"
 - func[5] <meshopt_decodeIndexBuffer> -> "meshopt_decodeIndexBuffer"
 - func[3] <sbrk> -> "sbrk"
```

Moreover, trying to add _start to the list of exported functions (via EXPORTED_FUNCTIONS) produces:

> shared:ERROR: undefined exported function: "_start"

**Comment by kripken** [comment]
Does that happen on 1.39.1? If so please provide a testcase if you can, as that sounds like an unknown bug (that is missed in our testing somehow), `_start` should always exist.

**Comment by zeux** [comment]
Yeah this is in 1.39.1. Maybe I'm missing something.

```c++
static int data[1024];

static int buildData()
{
        for (int i = 0; i < 1024; ++i)
        {
                data[i] = i * i;
        }

        return 0;
}

static int global = buildData();

extern "C" int bar(int x)
{
        return data[x];
}
```

```
$ emcc test.cpp -O3 -s EXPORTED_FUNCTIONS='["_bar"]' -s STANDALONE_WASM=1 -o test.wasm
$ ls -la test.wasm
-rwxrwxrwx 1 zeux zeux 76 Nov  2 16:27 test.wasm
$ wabt/wasm-objdump -x test.wasm

test.wasm:      file format wasm 0x1

Section Details:

Type[1]:
 - type[0] (i32) -> i32
Function[1]:
 - func[0] sig=0 <bar>
Memory[1]:
 - memory[0] pages: initial=256 max=256
Export[2]:
 - memory[0] -> "memory"
 - func[0] <bar> -> "bar"
Code[1]:
 - func[0] size=14 <bar>
Data[1]:
 - segment[0] memory=0 size=3 - init i32=5632
  - 0001600: a016 50                                  ..P
$ wabt/wasm-objdump -d test.wasm

test.wasm:      file format wasm 0x1

Code Disassembly:

000032 func[0] <bar>:
 000033: 20 00                      | local.get 0
 000035: 41 02                      | i32.const 2
 000037: 74                         | i32.shl
 000038: 41 80 08                   | i32.const 1024
 00003b: 6a                         | i32.add
 00003c: 28 02 00                   | i32.load 2 0
 00003f: 0b                         | end
```

**Comment by sbc100** [comment]
I was able to repro this.  I looks like we should always include the _start
symbol rather then only including it when main exists.   This is because
start has two purposes, to call main but also to call static constructors.

I'll open a PR to fix this.

**Comment by zeux** [comment]
Fixed by #9771. Thanks!
